server/lorax_server/models/__init__.py

```diff
@@ -72,15 +73,21 @@ def get_model(
         raise ValueError(f"Unknown source {source}")
 
     model_type = config_dict["model_type"]
+    dtype = dtype or config_dict.get("torch_dtype", "float16")
 
-    if dtype is None:
-        dtype = torch.float16
-    elif dtype == "float16":
+    if dtype in {"float16", "float32"}:
         dtype = torch.float16
     elif dtype == "bfloat16":
-        dtype = torch.bfloat16
+        if is_bf16_supported():
+            dtype = torch.bfloat16
+        else:
+            logger.warning("bfloat16 is not supported on this device, falling back to float16")
+            dtype = torch.float16
     else:
-        raise RuntimeError(f"Unknown dtype {dtype}")
+        try:
```

Comment:
If the user explicitly specified bf16 but it failed, that seems like it should be an error

Reply:
Agreed